### PR TITLE
DEV 015 sorting searchresults

### PR DESCRIPTION
**Pull request by Paul-Gerarts**
Please test this branch functionally as well. Flow should be as accurate as described.

### CODE REVIEW

File: src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java [hunk 1]
```diff
@@ -9,48 +9,128 @@
...
+        List<Product> productList = productService.findAllArchived().getProducts();
+        return getProductsSortedByPrice(model, productList);
+    }
+
+    private List<Product> applySearch(boolean isSearchResultView) {
+        searchService.setSearchResultView(isSearchResultView);
+        searchService.setArchivedView(false);
+        List<Product> productList = productListCacheService.findBySearchRequest(searchService.getSearchModel()).getProducts();
+        List<Product> filteredList = productListCacheService.filterByPrice(productList, searchService.getSearchModel()).getProducts();
+        return (null != searchService.getSearchModel().getDescription())
```
Comment (Fonthom) [inline src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java:102]: boolean hasProductDescription() is a more readable if statement.

voorbeeld: bindingResult.hasErrors() -> same solution for a null-check.

File: src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java [hunk 2]
```diff
@@ -46,14 +59,103 @@ private boolean checkIfProductsCacheNeedsUpdate() {
...
+    private ProductList getSearchResultsOrAllProducts(List<Product> result) {
+        searchService.setSearchFailure(result.isEmpty());
+        return result.isEmpty()
+                ? new ProductList(getProductListCache().getProducts())
+                : new ProductList(result);
+    }
+
+    private void setAppliedFiltersToSearchModel(SearchModel searchModel) {
+        searchModel.setAppliedFiltersHeader(" with the applied filters");
+        String searchRequest = (null != searchService.getSearchModel().getSearchRequest())
```
Comment (Fonthom) [inline src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java:103]: boolean hasSearchRequest() is a more readable name for this if statement.

File: src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java [hunk 3]
```diff
@@ -46,14 +59,103 @@ private boolean checkIfProductsCacheNeedsUpdate() {
...
+    private ProductList getSortedList(List<Product> products, Comparator<Product> productComparator) {
+        return new ProductList(
+                products
+                        .stream()
+                        .sorted(productComparator)
+                        .collect(Collectors.toUnmodifiableList()));
+    }
+
+    private List<Product> executeSearch(String searchRequest) {
+        return (null != searchRequest)
```
Comment (Fonthom) [inline src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java:140]: boolean hasSearchRequest(searchRequest) is more readable in this if statement.

File: src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java [hunk 4]
```diff
@@ -46,14 +59,103 @@ private boolean checkIfProductsCacheNeedsUpdate() {
...
                         .toLowerCase()
                         .contains(searchRequest.toLowerCase()))
-                .collect(Collectors.toUnmodifiableList());
-        return result.isEmpty() ? new ProductList(getProductListCache().getProducts()) : new ProductList(result);
+                .collect(Collectors.toUnmodifiableList())
+                : new ArrayList<>();
+    }
+
+    private List<Product> executeDescriptionSearch(List<Product> products, String description) {
+        return (null != description)
```
Comment (Fonthom) [inline src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java:151]: boolean hasDescription(description) is more readable in this if statement.

File: src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java (hunk 1 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java:102]: extracted

File: src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java (hunk 2 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java:103]: extracted

File: src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java (hunk 3 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java:140]: I disagree. Whilst I get the logic behind extracting every little boolean-check, I myself wouldn't do so for simple null-checks, especially this one. Here's why: 
Above, I've extracted a method checking for searchRequest. Hence, the method already exists. Doing the exact same thing here, would indeed result in the method you suggest. BUT, that would mean that my previous extracted boolean-check would become something like -> hasSearchRequest(searchService.getSearchModel().getSearchRequest()).
If we're looking for increased readability, this isn't the way to go...

File: src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java (hunk 4 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/services/ProductListCacheServiceImpl.java:151]: see explanation for the searchRequest one...

**Pull request merged**